mkt/extensions/tests/test_views.py

```diff
@@ -716,6 +716,119 @@ def test_search_author(self):
+        response = self.anon.get(self.url)
+        eq_(response.status_code, 403)
+
+    def test_list(self):
+        self.extension2 = Extension.objects.create(name=u'Mŷ Second Extension')
+        self.version2 = ExtensionVersion.objects.create(
+            extension=self.extension2, status=STATUS_PUBLIC, version='1.2.3')
+        self.refresh('extension')
+        response = self.client.get(self.url)
+        eq_(response.status_code, 200)
+        eq_(len(response.json['objects']), 2)
+
+    def test_query(self):
+        self.extension2 = Extension.objects.create(name=u'Superb Extensiôn')
+        self.version2 = ExtensionVersion.objects.create(
+            extension=self.extension2, status=STATUS_PUBLIC, version='4.5.6')
+        self.refresh('extension')
+        res = self.client.get(self.url, data={'q': 'superb'})
+        eq_(res.status_code, 200)
+        objs = res.json['objects']
+        eq_(len(objs), 1)
+        eq_(objs[0]['id'], self.extension2.pk)
+
+    def test_query_sort_reviewed(self):
+        self.extension2 = Extension.objects.create(name=u'Superb Extensiôn')
+        self.version2 = ExtensionVersion.objects.create(
+            extension=self.extension2, reviewed=self.days_ago(0),
+            status=STATUS_PUBLIC, version='4.5.6')
+        self.refresh('extension')
+        res = self.client.get(
+            self.url, data={'q': 'extension', 'sort': 'reviewed'})
+        eq_(res.status_code, 200)
+        objs = res.json['objects']
+        eq_(len(objs), 2)
+        eq_(objs[0]['id'], self.extension2.pk)
+        eq_(objs[1]['id'], self.extension.pk)
+
+    def test_query_sort_relevance(self):
+        self.extension2 = Extension.objects.create(
+            name=u'Superb Extensiôn', slug='superb-lol')
+        self.version2 = ExtensionVersion.objects.create(
+            extension=self.extension2, reviewed=self.days_ago(0),
+            status=STATUS_PUBLIC, version='4.5.6')
+        self.refresh('extension')
+        res = self.client.get(self.url, data={'q': 'extension'})
+        eq_(res.status_code, 200)
+        objs = res.json['objects']
+        eq_(len(objs), 2)
+        eq_(objs[0]['id'], self.extension.pk)
+        eq_(objs[1]['id'], self.extension2.pk)
+
+    def test_query_no_results(self):
+        res = self.client.get(self.url, data={'q': 'something'})
+        eq_(res.status_code, 200)
+        eq_(res.json['objects'], [])
+
+    def test_not_public(self):
+        self.extension.update(status=STATUS_PENDING)
+        self.refresh('extension')
+        response = self.client.get(self.url)
+        eq_(response.status_code, 200)
+        eq_(len(response.json['objects']), 1)
+
+    def test_disabled(self):
+        self.extension.update(disabled=True)
+        self.refresh('extension')
+        response = self.client.get(self.url)
+        eq_(response.status_code, 200)
+        eq_(len(response.json['objects']), 1)
+
+    def test_deleted(self):
+        self.extension.update(deleted=True)
```

Comment:
Ok. We just need to make sure this is clearly marked as deleted. Also, the detail endpoint does not allow you to see deleted instances yet.